src/analysis/inner_product_space/of_norm.lean

```diff
@@ -0,0 +1,407 @@
+/-
+Copyright (c) 2020 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+
+import analysis.normed_space.lattice_ordered_group
+import topology.algebra.algebra
+import analysis.inner_product_space.basic
+
+/-!
+# Inner product space derived from a norm
+
+This file defines an `inner_product_space` instance from a norm that respects the
+parallellogram identity.
```

Comment:
Could you recall here what the parallelogram identity is? And maybe sketch the proof you will be giving somewhere in the file-level docstring, with pointers to intermediate lemmas in the proof.